File: Chat/Chat/wrapper/java/JChatSDK/src/com/ringid/imsdk/DataCounter.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.ringid.imsdk;

/**
 *
 * @author towhid
 */
public class DataCounter {
    private long dataSent;
    private long dataReceived;

    public DataCounter(){
        
    }
    
    public DataCounter(long dataSent, long dataReceived){
        this.dataSent = dataSent;
        this.dataReceived = dataReceived;
    }
    /**
     * @return the dataSent
     */
    public long getDataSent() {
        return dataSent;
    }

    /**
     * @return the dataReceived
     */
    public long getDataReceived() {
        return dataReceived;
    }
}

File: Chat/ChatParser/src/DataCounter.cpp
#include "DataCounter.h"

namespace imsdk{

    DataCounter::DataCounter(long long dataSent, long long dataReceived) {
        this->dataSent = dataSent;
        this->dataReceived = dataReceived;
    }
        DataCounter::DataCounter() {
    }
    
    DataCounter::~DataCounter(){
        
    }
    
    void DataCounter::setDataReceived(long long dataReceived) {
        this->dataReceived = dataReceived;
    }

    long long DataCounter::getDataReceived() const {
        return dataReceived;
    }

    void DataCounter::setDataSent(long long dataSent) {
        this->dataSent = dataSent;
    }

    long long DataCounter::getDataSent() const {
        return dataSent;
    }
    
    void DataCounter::addDataReceived(long long dataReceived) {
        this->dataReceived += dataReceived;
    }
    
    void DataCounter::addDataSent(long long dataSent) {
        this->dataSent += dataSent;
    }
}
